client/src/components/matchChat/MatchChat.tsx

```diff
@@ -0,0 +1,327 @@
+import React, {
+  useEffect,
+  useState,
+  ChangeEvent,
+  KeyboardEvent,
+  useRef,
+  useContext,
+} from 'react';
+import InsertEmoticonIcon from '@mui/icons-material/InsertEmoticon';
+import AttachFileIcon from '@mui/icons-material/AttachFile';
+import SendIcon from '@mui/icons-material/Send';
+import CallOutlinedIcon from '@mui/icons-material/CallOutlined';
+import VideocamOutlinedIcon from '@mui/icons-material/VideocamOutlined';
+import Typography from '@mui/material/Typography';
+import axios, { AxiosError } from 'axios';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { Box } from '@mui/material';
+import data from '@emoji-mart/data';
+import Picker from '@emoji-mart/react';
+import {
+  AddMessageBar,
+  IconBackground,
+  MessageInput,
+  Wrapper,
+} from './MatchChat.styled';
+import Message from './Message';
+import { CustomErrorResponse, IMatchDataProps } from '../../interfaces';
+
+import ChatImage from '../../assets/chat.svg';
+import { AuthContext } from '../../context';
+import ErrorAlert from '../ErrorAlert';
+
+const MatchChat = () => {
+  const { pathname } = useLocation();
+  const matchId = Number(pathname.split('/')[3]);
+
+  const [matchData, setMatchData] = useState<IMatchDataProps>({
+    status: 0,
+    data: {
+      match: {
+        MatchMessages: [],
+        createdAt: '',
+        description: '',
+        endDate: '',
+        id: 0,
+        ownerId: 0,
+        seats: 0,
+        stadiumId: 0,
+        startDate: '',
+        title: '',
+        updatedAt: '',
+      },
+    },
+  });
+
+  const [messageInput, setMessageInput] = useState<string>('');
+  const [newMessage, setNewMessage] = useState<object | null>(null);
+  const [isIconPickerShown, setIsIconPickerShown] = useState<boolean>(false);
+  const [isDeleted, setIsDeleted] = useState<object | null>(null);
+
+  const [errorMessage, setErrorMessage] = useState<string>('');
+
+  const scrollContainerRef = useRef<HTMLDivElement>();
+
+  const navigate = useNavigate();
+
+  const { user } = useContext(AuthContext);
+
+  const matchMessages = matchData?.data?.match?.MatchMessages;
+
+  const handleScrollChat = () => {
+    if (scrollContainerRef.current) {
+      scrollContainerRef.current.scrollTo({
+        top: scrollContainerRef.current.scrollHeight,
+        behavior: 'smooth',
+      });
+    }
+  };
+
+  function isAxiosError(error: unknown): error is AxiosError {
+    return (error as AxiosError).isAxiosError !== undefined;
+  }
+
+  useEffect(() => {
+    handleScrollChat();
```

Comment:
what is this :
so to see other messages I want to send a new message
this is not cooked

Reply:
You misunderstood, it will scroll to the bottom when you send a new message.

Reply:
sorry @ahmedirheem  but this request change for the next line